### PR TITLE
Apply astyle to format all code (v3)

### PR DESCRIPTION
**Pull request by dpgeorge**
Per discussion in #4223, here is a 3rd attempt at reformatting the code using astyle.

I've converted the `codeformat.py` script to Python so it works on Windows as well.

The aim with this attempt is to minimise the diff as much as possible, so some options that we do want to eventually enable in astyle's format are not yet enabled (eg remove `--keep-one-line-statements`).

This PR is down to about 3000 additions and 3000 deletions, compared to 5800/4700 for  attempt 1, and 4600/4600 for attempt 2.

But still there are some things which are not great, mainly to do with handling of preprocessor lines.  Eg look at `py/nlr.h`, the #if' block that configures `MICROPY_NLR_NUM_REGS` is really unreadable now.

Unless we make a custom version of astyle, I don't think it'll ever be perfect, so at some point concessions must be made.  Just need to decide what those concessions are.

### CODE REVIEW

**Comment by robert-hh** [comment]
IMHO, applying astyle is in any case a big bang. So the focus should be on clean, safe and easier-to-maintain code. If that can be achieved, minimizing the number of changed lines seems less important to me.

**Comment by dlech** [comment]
It is also possible to use `// *INDENT-OFF*` and `// *INDENT-ON*` to temporarily disable and enable the formatter for a section of code. This could be used sparingly for places like `py/nlr.h`.

There is also `// *NOPAD *` to work around issues described in https://github.com/micropython/micropython/pull/5631#issuecomment-586024256

http://astyle.sourceforge.net/astyle.html#_Disable_Formatting

File: extmod/machine_signal.c [hunk 1]
```diff
@@ -93,7 +93,7 @@ STATIC mp_obj_t signal_make_new(const mp_obj_type_t *type, size_t n_args, size_t
     }
     else
     #endif
-    // Otherwise there should be 1 or 2 args
+        // Otherwise there should be 1 or 2 args
```
Comment (Jongy) [inline extmod/machine_signal.c]: This look wrong

**Comment by Jongy** [comment]
> IMHO, applying astyle is in any case a big bang. So the focus should be on clean, safe and easier-to-maintain code. If that can be achieved, minimizing the number of changed lines seems less important to me.

Agreed.

I've taken a short look. It looks like there are many misses with the positioning of `#ifdef`s. MicroPython has a very... peculiar convention for them, I don't recall any other project I've seen with the same positioning, so I don't think it'll be easy to get it right.

Anyway, I think autoformatting is a great improvement for this project, and if any help is required, I'd be happy in join in the forces...

**Comment by MrSurly** [comment]
What about having a server-side hook to check style?

**Comment by jimmo** [comment]
> What about having a server-side hook to check style?

+1 -- would be keen to see this as a github hook like the existing travis/appveyor/coverage. But it think it's important that it only looks at the patch, or at most the changed files.

For clang-format, there's also clang-format-diff.py which provides a way to run clang-format on just the diff from commits which would be useful for making such a hook. Does astyle have a similar feature?

**Comment by stinos** [comment]
> But it think it's important that it only looks at the patch, or at most the changed files.

Can you elaborate why? Without thinking too much about it, I thought the consensus is 'entire codebase should be formatted' in which case it doesn't matter that the Travis builds comparse all of it or just the patch

**Comment by jimmo** [comment]
> Can you elaborate why? Without thinking too much about it, I thought the consensus is 'entire codebase should be formatted' in which case it doesn't matter that the Travis builds comparse all of it or just the patch

If I submit a PR with correct formatting, then I shouldn't see an error because of existing unrelated formatting issues.

Conversely, if I were a new contributor and the PR showed up with an error due to (unrelated) formatting, that would be weird and confusing (and we shouldn't give people the idea that they should need to go and fix unrelated things in their PR).

But I agree that in practice this shouldn't happen...

**Comment by stinos** [comment]
> existing unrelated formatting issues.

Ah. But the thing is: there shouldn't be such issues anymore right?

File: extmod/modbluetooth.c [hunk 2]
```diff
@@ -852,7 +852,7 @@ STATIC mp_obj_t bluetooth_ble_invoke_irq(mp_obj_t none_in) {
         } else if (event == MP_BLUETOOTH_IRQ_GATTS_WRITE) {
             // conn_handle, value_handle
             ringbuf_extract(&o->ringbuf, data_tuple, 2, 0, NULL, 0, 0, NULL, NULL);
-        #if MICROPY_PY_BLUETOOTH_ENABLE_CENTRAL_MODE
+            #if MICROPY_PY_BLUETOOTH_ENABLE_CENTRAL_MODE
         } else if (event == MP_BLUETOOTH_IRQ_SCAN_RESULT) {
             // addr_type, addr, connectable, rssi, adv_data
```
Comment (codefreax) [inline extmod/modbluetooth.c]: This is hard for a formatter to get right, and that may be an argument to fully outdent PP statements.

**Comment by codefreax** [comment]
```
self->data[uuid_i/2] = c << 4;
```
Is it MP style to not have spaces around operators if inside square brackets? If not, this might be the opportunity to make it consistent.

**Comment by dhalbert** [comment]
Does `--indent-preproc-block` help? (http://astyle.sourceforge.net/astyle.html#_indent-preproc-block). I see you have `--indent-preproc-cond`, but not `-block`.

File: extmod/modbluetooth.c (hunk 2 above)
Comment (dpgeorge) [reply extmod/modbluetooth.c]: Yes, it's really hard to get #'s correct.  Probably the best thing for the long run is to just not indent #'s at all.

**Comment by dpgeorge** [comment]
> Is it MP style to not have spaces around operators if inside square brackets? If not, this might be the opportunity to make it consistent.

The style is to have spaces around all operators.  The clang-format version does fix this.

**Comment by dpgeorge** [comment]
> Does `--indent-preproc-block`

No that doesn't help, that is not a very good option to use because it indents everything (including code) within #if's, eg:
```diff
 #if MICROPY_PY_BUILTINS_FLOAT
-mp_float_t mp_obj_get_float(mp_obj_t self_in);
-bool mp_obj_get_float_maybe(mp_obj_t arg, mp_float_t *value);
-void mp_obj_get_complex(mp_obj_t self_in, mp_float_t *real, mp_float_t *imag);
+    mp_float_t mp_obj_get_float(mp_obj_t self_in);
+    bool mp_obj_get_float_maybe(mp_obj_t arg, mp_float_t *value);
+    void mp_obj_get_complex(mp_obj_t self_in, mp_float_t *real, mp_float_t *imag);
 #endif
```

**Comment by dpgeorge** [comment]
Closed in favour of #5700.

**Pull request closed without merging**